procurement_purchase_no_grouping/models/purchase_order.py

```diff
@@ -29,10 +29,11 @@ def _find_candidate(
         modules, and on standard operation mode, procurement groups are not reused
         between sales orders.
         """
-        obj = self
         if values.get("grouping") == "line":
-            obj = self.filtered(lambda x: x.order_id.group_id == values.get("group_id"))
-        return super(PurchaseOrderLine, obj)._find_candidate(
+            self = self.filtered(
+                lambda x: x.order_id.group_id == values.get("group_id")
```

Comment:
values.get("group_id") is an id or recordset??

Reply:
It is a recordset (https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_move.py#L1405)

Reply:
Odoo putting very badly the variable names...